### ISSUE TITLE
interaction with geometry package

### ISSUE THREAD

**Issue by davidcarlisle** [texjporg/uplatex#5]
The stackexchange posting at 

https://tex.stackexchange.com/questions/522069/uptex-utarticle-class-and-geometry-package-and-setup

seems to suggest that the geometry package and utarticle class don't really work well together,

This is mainly an issue with geometry not understanding the writing direction modes of ptex (I think) but  I am reporting it here rather than at 

https://github.com/davidcarlisle/geometry

as that just comes to me, and my understanding of ptex writing modes is rather limited.

Any suggestions welcome (or if my stackexchange answer is completely wrong feel free to edit or ask me to delete)

**Comment by aminophen** [comment]
The fact that geometry does not support vertical writing classes is well-known in Japan; the workaround for the issue is to use lltjp-geometry.sty provided by luatexja project team. I will add an answer in tex.sx later.

**Comment by aminophen** [comment]
Do you think it's possible/better to sort the hooks in lltjp-geometry.sty to be merged into the mainstream geometry.sty? > @h-kitagawa

**Comment by davidcarlisle** [comment]
@aminophen  yes I think it's always better to fix an upstream package than have a package that patches another.

I can't really do this though as while I more or less understand the vertical mode, I don't have a feeling for the expected affect on the global page layout description. Also to coordinate with  lltjp-geometry.sty so that any changes in geometry don't break existing documents using that.

If it requires major changes should check that Hideo Umeki  agrees. I set up the geometry github repository for some "emergency fixes" when luatex changes broke geometry and  Hideo Umeki  had no time to fix, but I feel I am only temporarily holding the  package.

**Comment by h-kitagawa** [comment]
> Do you think it's possible/better to sort the hooks in lltjp-geometry.sty to be merged into the mainstream geometry.sty? > @h-kitagawa

I think that it's not easy, because of how to detect writing direction (horizontal or vertical). I hope the present approach (looking the writing direction at loading the package, and checking that there is \tate in \@begindocumenthook) works in almost all cases, but I don't think it covers every cases.

**Comment by davidcarlisle** [comment]
> I think that it's not easy, because of how to detect writing direction (horizontal or vertical). 

one possibility would be to not auto-detect but simply have an option (`tate` or `vertical` or something?) that the user could use to enable this.

in fact I think one thing that could probably be done straight away is to add a `tate` option to geometry that  simply does `\RequirePackage{ lltjp-geometry}` so that the user just needs to add `[tate]` (which would be documented in the geometry manual), as an alternative  to changing `\usepackage{geometry}` to `\usepackage{lltjp-geometry}`  does this sound reasonable?

Longer term  I think it would be good to combine the packages but there is no rush to change anything this has clearly been this way for a while.

**Comment by aminophen** [comment]
For the record: `lltjp-geometry` is not working after LaTeX2e 2020-02-02.

* Original report: https://oku.edu.mie-u.ac.jp/tex/mod/forum/discuss.php?d=2781
* Related: https://github.com/latex3/latex3/issues/681
* Related: https://ja.osdn.net/projects/luatex-ja/ticket/40058

-> edit: fixed in luatexja 20200227.0
